### PR TITLE
274 add multi organ segmentation bundle

### LINKED ISSUE #274: Add multi-organ segmentation bundle
Provide multi-organ segmentation bundle for abdominal CT images.

### PR DESCRIPTION
**Pull request by cshen0625**
Fixes #274

### Description
Add multi-organ segmentation bundle for abdominal CT images.

### Status
**Ready**

### Please ensure all the checkboxes:

- [x] Codeformat tests passed locally by running ./runtests.sh --codeformat.
- [ ] In-line docstrings updated.
- [x] Update `version` and `changelog` in `metadata.json` if changing an existing bundle.
- [x] Please ensure the naming rules in config files meet our requirements (please refer to: `CONTRIBUTING.md`).
- [x] Ensure versions of packages such as `monai`, `pytorch` and `numpy` are correct in `metadata.json`.
- [x] Descriptions should be consistent with the content, such as `eval_metrics` of the provided weights and TorchScript modules.
- [x] Files larger than 25MB are excluded and replaced by providing download links in `large_file.yml`.
- [x] Avoid using path that contains personal information within config files (such as use `/home/your_name/` for `"bundle_root"`).

### CODE REVIEW

File: multi_organ_segmentation/docs/README.md
Comment (yiheng-wang-nv) [inline multi_organ_segmentation/docs/README.md]: Hi @cshen0625 , thanks for contributing this bundle! As for the dataset mentioned here, could you please also attach the way of how to get it (or if it is private, also need to mention here, like [this bundle does](https://github.com/Project-MONAI/model-zoo/tree/dev/models/endoscopic_tool_segmentation#data))

Comment (cshen0625) [reply multi_organ_segmentation/docs/README.md]: Hi @yiheng-wang-nv , thank you for your comment. This dataset is a private data. I will update the README and submit it again.

Comment (yiheng-wang-nv) [reply multi_organ_segmentation/docs/README.md]: Thanks! and the file `docs/license.txt` can be removed as the dataset is not provided (like [this bundle](https://github.com/Project-MONAI/model-zoo/tree/dev/models/ventricular_short_axis_3label/docs))

Comment (cshen0625) [reply multi_organ_segmentation/docs/README.md]: Sure. I updated the bundle based on your comments.

File: multi_organ_segmentation/docs/README.md
Comment (yiheng-wang-nv) [inline multi_organ_segmentation/docs/README.md]: Here could the format of labels be changed? It may not need to use a dict based form in readme, thanks!
An example of the brats bundle is: https://github.com/Project-MONAI/model-zoo/tree/dev/models/brats_mri_segmentation#output

File: multi_organ_segmentation/docs/README.md
Comment (yiheng-wang-nv) [reply multi_organ_segmentation/docs/README.md]: Thanks! As for the bundle itself, does it have any license? Now the`LICENSE` file is empty

File: multi_organ_segmentation/docs/README.md
Comment (cshen0625) [reply multi_organ_segmentation/docs/README.md]: I revised this problem.

**Comment by yiheng-wang-nv** [comment]
Hi @cshen0625 , I don't have other comments except the following two places:
1. may need to fix the code format error (https://github.com/Project-MONAI/model-zoo/actions/runs/3898889327/jobs/6658043327)
2. need to double confirm the LICENSE question. See: https://github.com/Project-MONAI/model-zoo#license   If not have any license, we suggest to use apache 2.0 license, please refer [this bundle](https://github.com/Project-MONAI/model-zoo/blob/dev/models/brats_mri_segmentation/LICENSE), and describe the license within the readme file (see: https://github.com/Project-MONAI/model-zoo/tree/dev/models/brats_mri_segmentation#license)
Thanks!

**Comment by yiheng-wang-nv** [comment]
> Hi @cshen0625 , I don't have other comments except the following two places:
> 
> 1. may need to fix the code format error (https://github.com/Project-MONAI/model-zoo/actions/runs/3898889327/jobs/6658043327)
> 2. need to double confirm the LICENSE question. See: https://github.com/Project-MONAI/model-zoo#license   If not have any license, we suggest to use apache 2.0 license, please refer [this bundle](https://github.com/Project-MONAI/model-zoo/blob/dev/models/brats_mri_segmentation/LICENSE), and describe the license within the readme file (see: https://github.com/Project-MONAI/model-zoo/tree/dev/models/brats_mri_segmentation#license)
>    Thanks!

I've discussed with Holger, and for the first error, I will try to fix it later. Don't worry about it

**Comment by yiheng-wang-nv** [comment]
The pytype issue is not related to this bundle, I just fixed it in #298

**Review by yiheng-wang-nv — APPROVED** [review]
Thanks! Look good to me. Hi @cshen0625 , could you please also copy the content of apache 2.0 license into the file `LICENSE`? Like in:
https://github.com/Project-MONAI/model-zoo/blob/dev/models/brats_mri_segmentation/LICENSE
Then I think this PR is ready

**Comment by yiheng-wang-nv** [comment]
Since it is not updated, I will push a commit to address the license content issue, and then merge this PR

**Comment by holgerroth** [comment]
@yiheng-wang-nv , please wait a bit. We are waiting for approval from the university for the License text.

**Comment by yiheng-wang-nv** [comment]
> @yiheng-wang-nv , please wait a bit. We are waiting for approval from the university for the License text.

Hi @holgerroth , I see, sure. The commits I submitted are to fix a few type errors.

**Comment by Nic-Ma** [comment]
Hi @holgerroth ,

Any update for the approval from the university for the License text?

Thanks.

**Comment by holgerroth** [comment]
> 

@cshen0625, can you share any updates?

**Comment by cshen0625** [comment]
> > 
> 
> @cshen0625, can you share any updates?




@holgerroth Sorry for waiting, I update the license.

File: models/multi_organ_segmentation/configs/train.yaml
Comment (holgerroth) [inline models/multi_organ_segmentation/configs/train.yaml]: Needs to follow the "initialize", "run", "finalize" schema.

**Review by holgerroth — CHANGES_REQUESTED** [review]
Configs need to follow the "initialize", "run", "finalize" schema.

File: models/multi_organ_segmentation/configs/train.yaml
Comment (cshen0625) [reply models/multi_organ_segmentation/configs/train.yaml]: Update the configs based on the latest schema.

File: LICENSE
Comment (holgerroth) [inline LICENSE:1]: @cshen0625 , I think you changed this root license file by mistake. Please restore the original version.

**Comment by holgerroth** [comment]
@Nic-Ma , @yiheng-wang-nv The license has been updated and configs are updated to the latest monai version. Tested with 1.2.0rc6. I think this PR can be merged now.

**Comment by Nic-Ma** [comment]
@yiheng-wang-nv Please review it again ASAP when you are available.

Thanks.

**Comment by holgerroth** [comment]
We might have to retrain this model as it was trained with 1.0.1. I tested the TRT inference but it turns out the evaluation gives some key mismatch when trying to load the model checkpoint. @cshen0625 , please let us know if you can retrain the model using the 1.2.0 release candidate.

File: models/multi_organ_segmentation/configs/metadata.json
Comment (yiheng-wang-nv) [inline models/multi_organ_segmentation/configs/metadata.json:16]: Hi @holgerroth @cshen0625 , we decided to add a new key "name" in metadata after my last review.
It is used as a display name of the bundle in the webpage.
Could you please help to add it? Thanks!

File: models/multi_organ_segmentation/configs/dataset_0.json
Comment (yiheng-wang-nv) [inline models/multi_organ_segmentation/configs/dataset_0.json:1]: As for this file, since the related dataset is private, do you think this file is necessary?
If the purpose is to shown what is the form of the data list, maybe you can refer to this bundle:
https://github.com/Project-MONAI/model-zoo/tree/dev/models/endoscopic_inbody_classification#preprocessing

File: models/multi_organ_segmentation/configs/metadata.json
Comment (Nic-Ma) [reply models/multi_organ_segmentation/configs/metadata.json:16]: @yiheng-wang-nv Please provide an example in some bundle for reference.

Thanks.

Comment (yiheng-wang-nv) [reply models/multi_organ_segmentation/configs/metadata.json:16]: here is an example:
https://github.com/Project-MONAI/model-zoo/blob/b72b9fbe80e999407d4d136207778a02f37d3c6d/models/brats_mri_axial_slices_generative_diffusion/configs/metadata.json#L18

**Comment by Nic-Ma** [comment]
Hi @cshen0625 @holgerroth ,

May I know the status and plan for this PR?

Thanks.

**Comment by holgerroth** [comment]
> Hi @cshen0625 @holgerroth ,
> 
> May I know the status and plan for this PR?
> 
> Thanks.

@cshen0625 is currently retraining this model with a 1.2 rc as we ran into compatibility issues with the old checkpoints. Should be done soon.

**Comment by cshen0625** [comment]
@holgerroth @Nic-Ma 
Hi, I updated the bundle using the new MONAI version.

**Comment by yiheng-wang-nv** [comment]
Thanks for the updates @cshen0625 , I will review it again soon!

File: models/multi_organ_segmentation/configs/metadata.json
Comment (yiheng-wang-nv) [inline models/multi_organ_segmentation/configs/metadata.json]: Hi @cshen0625 , here could you use `"pytorch_version": "1.13.1"`? We may need a formal torch version (rather than a dev version).
In addition, I suggest to use `"monai_version": "1.2.0"`

Comment (cshen0625) [reply models/multi_organ_segmentation/configs/metadata.json]: Sure. I will test if it works with 1.2.0.
I used the MONAI container with tag 1.2.0rc7, which was the latest version available at that time.

Comment (cshen0625) [reply models/multi_organ_segmentation/configs/metadata.json]: Hi @yiheng-wang-nv , this bundle works with `"monai_version": "1.2.0"` and `"pytorch_version": "1.13.1"`. Can I just modify the information in `metadata.json`?

Comment (yiheng-wang-nv) [reply models/multi_organ_segmentation/configs/metadata.json]: Yes @cshen0625 . Thanks!

Comment (cshen0625) [reply models/multi_organ_segmentation/configs/metadata.json]: Hi @yiheng-wang-nv , I updated the `metadata.json` file.

File: models/multi_organ_segmentation/large_files.yml
Comment (yiheng-wang-nv) [inline models/multi_organ_segmentation/large_files.yml]: Hi @cshen0625 , could you please also help to check the weights and hash values?
The CI test is failed and it seems the hash value is wrong. I tried to download `model.pt` locally and checked `md5sum`, and I got: `e21d663b33d29a8ca4e28d60a14a3d66`

Comment (cshen0625) [reply models/multi_organ_segmentation/large_files.yml]: Hi @yiheng-wang-nv , I'm sorry that I forgot to update the hash information. I have now updated the hash value.

File: models/multi_organ_segmentation/configs/metadata.json
Comment (yiheng-wang-nv) [reply models/multi_organ_segmentation/configs/metadata.json:16]: Hi @cshen0625 , could you please also address this?
We added the requirement of the "name" (see: https://github.com/Project-MONAI/model-zoo/blob/b245833e22544126f23925eac89da471562467a1/CONTRIBUTING.md#model-naming)

File: models/multi_organ_segmentation/configs/dataset_0.json
Comment (holgerroth) [reply models/multi_organ_segmentation/configs/dataset_0.json:1]: I think it's still good to have the file there for later reference. It documents what data was used to train this model.

File: models/multi_organ_segmentation/configs/metadata.json
Comment (holgerroth) [reply models/multi_organ_segmentation/configs/metadata.json:16]: Added ` "name": "Abdominal multi-organ segmentation",`

**Review by holgerroth — APPROVED** [review]

File: models/multi_organ_segmentation/docs/README.md
Comment (yiheng-wang-nv) [inline models/multi_organ_segmentation/docs/README.md]: Hi @cshen0625 @holgerroth , the license file shown in: https://github.com/Project-MONAI/model-zoo/pull/275/files#diff-009b21cc28e45820d20ee05f45a2c33421d542c2f932c6da8f66bd46990c02f4R1 is different from this section. Should one of them be modified?

Comment (holgerroth) [reply models/multi_organ_segmentation/docs/README.md]: This was updated.

**Pull request merged**
